### DISCUSSION TITLE
Yaml specs for the schema defintions

### DISCUSSION THREAD

**Discussion by SanjayGautam** [cimug-org/CIMTool#139] [Q&A]
Dear All , 

I see that we are able to generate JSON Schema form the OWL Profile , I want to generated the same Schema but in Yaml output , do you think we should create one project related with that and try to accomplish as more and more folks are planning to use Swagger for their APIs ??

**Comment by Sveino** [comment]
It should be [Yaml-ld](https://json-ld.github.io/yaml-ld/spec/). I am seeing Yaml more as human interface language than machine-to-machine. We are checking out the possibility to use [LinkML](https://linkml.io/linkml/) as a future replacement of the OWL Profile and we can then use LinkML to generate any schema and instance syntax. CIMTool can now export LinkML. See the repo https://github.com/cimug-org/cpsm-docs as proof-of-concept.

**Reply by SanjayGautam** [reply]
Thanks @Sveino  I will be looking at the LINK ML and will share my experience with it .

**Comment by ucaiug-admin** [comment]
> Dear All , 
> 
> 
> 
> I see that we are able to generate JSON Schema form the OWL Profile , I want to generated the same Schema but in Yaml output , do you think we should create one project related with that and try to accomplish as more and more folks are planning to use Swagger for their APIs ?? 
> 
> 
> 
> 

Hi Sanjay. On the heels of Svein's response and on a couple of the items he pointed out. If you download the most recent CIMTool 2.1.0 release from a month ago you will find a new builder that will generate a LinkML document off of the profile. This is an YAML. It is simply the first step toward some longer range goals, that Svein, myself, and others are exploring as time has permitted. As Svein pointed out LinkML opens up broader possibilities. I think it's safe to say that there will most definitely be changes to the current version of the LinkML generated, as we continue to discuss how to generate LinkML artifacts that sufficiently express relevant information that would make it useful across both the supported profile tools we use, as well as by Sparx EA itself.  

As far as OpenAPI, this is one of our focuses and actually is the basis for the current work happening for the next edition of IEC 61968-100 for enterprise integration. Specifically, research into and formalization of an approach and conventions for OpenAPI specifications for integration of IEC standards in the 61968 and 62325 series. The JSON builder you noted in your local CIMTool installation corresponds to a reference implementation we did based on a draft of IEC 62361-104. This standard then is a building block for subsequently formalizing OpenAPI for IEC 61968-100.  

Are you currently working on OpenAPI specifications at CAISO to support exchanging existing IEC standards-based messages?  

Todd

**Reply by SanjayGautam** [reply]
Yes , I am working currently to convert the current and new B2B SOAP APIs to Open Api based , I feel that we still want to use CIMTool , we use Enterprise Architect tool to generate the CMIs and use CIMtool , but I am also exploring options to just use EA tool to get everything done.
